core/model/src/main/java/org/eclipse/rdf4j/model/impl/SimpleLiteral.java

```diff
@@ -66,6 +71,9 @@
 	 *--------------*/
 
 	protected SimpleLiteral() {
+		plain = false;
+		simple = false;
+		lateInit = true;
 	}
```

Comment:
I wish we could remove this and make all the variables in this object final. Maybe we can deprecate this class and create a new one with everything being final to use for our own code.

Reply:
There's `GenericLiteral` in the model-api package that might be a good spot for this?